notebook/nbextensions.py

```diff
@@ -285,132 +663,451 @@ class InstallNBExtensionApp(JupyterApp):
+        if logger:
+            logger.warn("Out of date: %s" % dest)
+        return True
+    if logger:
+        logger.info("Up to date: %s" % dest)
+    return False
+
+
+def _maybe_copy(src, dest, logger=None):
+    """Copy a file if it needs updating.
+
+    Parameters
+    ----------
+
+    src : string
+        A path that should exist from which to copy a file
+    src : string
+        A path that might exist to which to copy a file
+    logger : Jupyter logger [optional]
+        Logger instance to use
+    """
+    if _should_copy(src, dest, logger=logger):
+        if logger:
+            logger.info("Copying: %s -> %s" % (src, dest))
+        shutil.copy2(src, dest)
+
+
+def _safe_is_tarfile(path):
+    """Safe version of is_tarfile, return False on IOError.
+
+    Returns whether the file exists and is a tarfile.
+
+    Parameters
+    ----------
+
+    path : string
+        A path that might not exist and or be a tarfile
+    """
+    try:
+        return tarfile.is_tarfile(path)
+    except IOError:
+        return False
+
+
+def _get_nbextension_dir(user=False, sys_prefix=False, prefix=None, nbextensions_dir=None):
+    """Return the nbextension directory specified
+
+    Parameters
+    ----------
+
+    user : bool [default: False]
+        Get the user's .jupyter/nbextensions directory
+    sys_prefix : bool [default: False]
+        Get sys.prefix, i.e. ~/.envs/my-env/share/jupyter/nbextensions
+    prefix : str [optional]
+        Get custom prefix
+    nbextensions_dir : str [optional]
+        Get what you put in
+    """
+    if sum(map(bool, [user, prefix, nbextensions_dir, sys_prefix])) > 1:
+        raise ArgumentConflict("cannot specify more than one of user, sys_prefix, prefix, or nbextensions_dir")
+    if user:
+        nbext = pjoin(jupyter_data_dir(), u'nbextensions')
+    elif sys_prefix:
+        nbext = pjoin(ENV_JUPYTER_PATH[0], u'nbextensions')
+    elif prefix:
+        nbext = pjoin(prefix, 'share', 'jupyter', 'nbextensions')
+    elif nbextensions_dir:
+        nbext = nbextensions_dir
+    else:
+        nbext = pjoin(SYSTEM_JUPYTER_PATH[0], 'nbextensions')
+    return nbext
+
+
+def _nbextension_dirs():
+    """The possible locations of nbextensions.
+
+    Returns a list of known base extension locations
+    """
+    return [
+        pjoin(jupyter_data_dir(), u'nbextensions'),
+        pjoin(ENV_JUPYTER_PATH[0], u'nbextensions'),
+        pjoin(SYSTEM_JUPYTER_PATH[0], 'nbextensions')
+    ]
+
+
+def _get_config_dir(user=False, sys_prefix=False):
+    """Get the location of config files for the current context
+
+    Returns the string to the enviornment
+
+    Parameters
+    ----------
+
+    user : bool [default: False]
+        Get the user's .jupyter config directory
+    sys_prefix : bool [default: False]
+        Get sys.prefix, i.e. ~/.envs/my-env/etc/jupyter
+    """
+    if user and sys_prefix:
+        raise ArgumentConflict("Cannot specify more than one of user or sys_prefix")
```

Comment:
Hey @bollwyvl @parente... with the latest commit putting user space first by default... I am getting this:

```
$ jupyter nbextension enable nb_conda --py --sys-prefix
Traceback (most recent call last):
  File "/home/damian/miniconda3/envs/notebook_dev/bin/jupyter-nbextension", line 9, in <module>
    load_entry_point('notebook', 'console_scripts', 'jupyter-nbextension')()
  File "/home/damian/miniconda3/envs/notebook_dev/lib/python3.5/site-packages/jupyter_core/application.py", line 267, in launch_instance
    return super(JupyterApp, cls).launch_instance(argv=argv, **kwargs)
  File "/home/damian/miniconda3/envs/notebook_dev/lib/python3.5/site-packages/traitlets/config/application.py", line 589, in launch_instance
    app.start()
  File "/media/data/devel/jupyter_project/notebook/notebook/nbextensions.py", line 909, in start
    super(NBExtensionApp, self).start()
  File "/home/damian/miniconda3/envs/notebook_dev/lib/python3.5/site-packages/jupyter_core/application.py", line 256, in start
    self.subapp.start()
  File "/media/data/devel/jupyter_project/notebook/notebook/nbextensions.py", line 833, in start
    self.toggle_nbextension_python(self.extra_args[0])
  File "/media/data/devel/jupyter_project/notebook/notebook/nbextensions.py", line 809, in toggle_nbextension_python
    logger=self.log)
  File "/media/data/devel/jupyter_project/notebook/notebook/nbextensions.py", line 451, in enable_nbextension_python
    logger=logger)
  File "/media/data/devel/jupyter_project/notebook/notebook/nbextensions.py", line 375, in _set_nbextension_state_python
    for nbext in nbexts]
  File "/media/data/devel/jupyter_project/notebook/notebook/nbextensions.py", line 375, in <listcomp>
    for nbext in nbexts]
  File "/media/data/devel/jupyter_project/notebook/notebook/nbextensions.py", line 332, in _set_nbextension_state
    _get_config_dir(user=user, sys_prefix=sys_prefix), 'nbconfig')
  File "/media/data/devel/jupyter_project/notebook/notebook/nbextensions.py", line 1043, in _get_config_dir
    raise ArgumentConflict("Cannot specify more than one of user or sys_prefix")
notebook.nbextensions.ArgumentConflict: Cannot specify more than one of user or sys_prefix
```

Can you verify that's the case?? If that's the case we should do something to allow --sys-prefix and make `user=False` (maybe at the flag setup level...)